dandi/dandiapi.py

```diff
@@ -1240,16 +1240,29 @@ def get_asset_by_path(self, path: str) -> RemoteAsset:
         paths (e.g., ``../``) within it.
         """
         path = self._normalize_path(path)
-        try:
-            # Weed out any assets that happen to have the given path as a
-            # proper prefix:
-            (asset,) = (
-                a for a in self.get_assets_with_path_prefix(path) if a.path == path
-            )
-        except ValueError:
-            raise NotFoundError(f"No asset at path {path!r}")
-        else:
-            return asset
+        # Weed out any assets that happen to have the given path as a
+        # proper prefix:
+        assets = [a for a in self.get_assets_with_path_prefix(path) if a.path == path]
+        if assets:
+            if len(assets) > 1:
+                lgr.warning(
+                    "Multiple assets found at path %r; returning the first one",
+                    path,
+                )
+            asset = assets[0]
+        elif not assets:
+            zarr_suf = ".zarr/"
+            if (zarr_suf in path) and (zarr_suffix_idx := path.index(zarr_suf)):
+                # If path ends with .zarr/, we might have a zarr asset without
+                # a trailing slash in the path.  Try again:
+                zarr_path_len = zarr_suffix_idx + len(zarr_suf)
+                zarr_path = path[: zarr_path_len - 1]  # -1 for trailing /
+                subpath = path[len(zarr_path) :]
```

Comment:
I have several issues with this:

- This code accepts paths containing a component that's just the Zarr suffix, e.g., `foo/.zarr/bar`.
- The behavior differs from dandidav, which tests each `.zarr`/`.ngff`-suffixed leading portion for Zarr-ness.
- Instead of futzing around with strings, I think it'd better to convert `path` to a `pathlib.PurePosixPath` and check `path.parts` to see if any part ends with `.zarr` (or `.ngff`, and also that the suffix is not the whole of the part).

Reply:
ok, agree that dealing with `Path` instance might be more robust